ray-on-gke/tpu/kuberay-tpu-webhook/main.go

```diff
@@ -47,16 +53,78 @@ var (
 	// headless svc will be of the form: {kuberay-cluster-name}-headless-worker-svc
 	headlessServiceSuffix = "headless-worker-svc"
 
-	// map of pod slices to workers in the slice
-	sliceToWorkers map[slice][]worker
-
 	// Flag arguments.
-	BindAddr   string
-	CACert     string
-	ServerCert string
-	ServerKey  string
+	BindAddr       string
+	CACert         string
+	KubeConfigPath string
+	ServerCert     string
+	ServerKey      string
 )
 
+func NewTPUWebhookServer(podLister listersv1.PodLister) *TPUWebhookServer {
+	return &TPUWebhookServer{
+		podLister: podLister,
+	}
+}
+
+func (t *TPUWebhookServer) Mutate(w http.ResponseWriter, r *http.Request) {
+	admissionReview := &admissionv1.AdmissionReview{}
+	if err := json.NewDecoder(r.Body).Decode(admissionReview); err != nil {
+		http.Error(w, "Error decoding request body", http.StatusBadRequest)
+		return
+	}
+
+	if admissionReview.Request.Kind.Kind == "Pod" {
```

Comment:
return StatusBadRequest if kind if not Pod?

Reply:
[dce1bbe] (https://github.com/GoogleCloudPlatform/ai-on-gke/pull/723/commits/dce1bbe4b09fd0a48304cb2d519414578a3ecdc0)

Reply:
I think we still want to return a BadRequest if the admission review is valid but the kind is not Pod
```
if admissionReview.Request.Kind.Kind != "Pod" {
     http.Error(w, "Invalid Kind", http.StatusBadRequest)
     w.WriteHeader(http.StatusBadRequest)
     return 
} 
```

Reply:
Added in daff1ed